src/tutorials/pet-shop.md

```diff
@@ -374,6 +376,100 @@ Since arrays can only return a single value given a single key, we create our ow
    ```
 
 Note the **memory** attribute on `adopters`. The memory attribute tells Solidity to temporarily store the value in memory, rather than saving it to the contract's storage. Since `adopters` is an array, and we know from the first adoption test that we adopted pet `expectedPetId`, we compare the testing contracts address with location `expectedPetId` in the array.
+</details>
+
+## Testing the smart contract using Javascript 
+
+<details>
+<summary> Expand This Section </summary>
+Truffle is very flexible when it comes to smart contract testing, in that tests can be written either in JavaScript or Solidity. In this tutorial, we'll be writing our tests in Javascript using chai and mocha library.<br/>
+
+1. Create a new file named `testAdoption.test.js` in the `test/` directory.<br/>
+2. Add the following content to the `testAdoption.test.js` file:<br/>
+
+  ```
+  const Adoption = require("../contracts/Adoption.sol");
+  
+  contract("Adoption", (accounts) => {
+    let adoption;
+    let expectedPetId;
+    
+    before(async() => {
+        adoption = await Adoption.deployed();
+    });
+
+    describe("adopting a pet and retrieving account addresses", async() => {
+      before("adopt a pet using accounts[0]", async () => {
+      await adoption.adopt(8, { from: accounts[0] });
+      expectedAdopter = accounts[0];
+    });
+    });
+  });
+
+  ```
+  We start the contract by importing : 
+  * ```Adoption.sol```:The smart contract we want to test
+  * ```chai```: Gives us various assertions to use in our tests to be used for equality, inequality or emptiness to return a pass/fail from our test.
+
+  **Note**: When writing the contract, we make use of the callback function ```accounts``` that provides us with the accounts in the development or the test network used.
+
+  Then, we make use of async functions : 
+  * To deploy the contract as soon as the test begins for the contract 
+  * To use the asynchronous feature to get the address of the contract and expectedAdopter
+
+  ### Testing the adopt() function 
```

Comment:
I think when talking about a function you should omit the "()" because that indicates a function call rather than the function itself. And you can probably use backticks (``) around your function name to style it. 

So like "Testing the `adopt` function"

Usually the backticks are used to indicate a variable or function name or something that is code.